onnxruntime/core/providers/tvm/tvm_so_execution_provider.cc

```diff
@@ -0,0 +1,301 @@
+                                               0, false};
+  allocator_ = CreateAllocator(default_memory_info);
+  InsertAllocator(allocator_);
+
+  // Get environment variables
+  const Env& env_instance = Env::Default();
+
+  const std::string dump_subgraphs_env = env_instance.GetEnvironmentVar(env_vars::kDumpSubgraphs);
+  if (!dump_subgraphs_env.empty()) {
+    LOGS(*GetLogger(), INFO) << "TVM EP processing shared lib does not save subgraphs " <<
+                                "due to it supports full model graph";
+  }
+}
+
+TvmSoExecutionProvider::~TvmSoExecutionProvider() {}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+TvmSoExecutionProvider::GetCapability(const GraphViewer& graph_viewer,
+                                      const std::vector<const KernelRegistry*>& /*kernel_registries*/) const {
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  if (graph_viewer.IsSubgraph()) {
+    return result;
+  }
+
+  const auto& init_tensors = graph_viewer.GetAllInitializedTensors();
+
+  std::unordered_set<std::string> required_initializers;
+  const std::vector<NodeIndex>& sorted_nodes = graph_viewer.GetNodesInTopologicalOrder();
+  std::unique_ptr<IndexedSubGraph> sub_graph = std::make_unique<IndexedSubGraph>();
+  for (auto& node_idx : sorted_nodes) {
+    graph_viewer.GetNode(node_idx)->ForEachDef([&required_initializers, &init_tensors]
+                                               (const NodeArg& node_arg, bool is_input) {
+              if(is_input && init_tensors.count(node_arg.Name())) {
+                  required_initializers.insert(node_arg.Name());
+              } }, true);
+  }
+
+  auto meta_def = std::make_unique<::onnxruntime::IndexedSubGraph::MetaDef>();
+  meta_def->name = "TVMStandalone";
+  meta_def->domain = "StandaloneTest";
+  std::vector<std::string> inputs;
+  std::vector<std::string> outputs;
+
+  for (auto& nodeArgPtr : graph_viewer.GetInputs()) {
+    inputs.push_back(nodeArgPtr->Name());
+  }
+
+  for (auto& name : required_initializers) {
+    inputs.push_back(name);
+  }
+
+  for (auto& nodeArgPtr : graph_viewer.GetOutputs()) {
+    outputs.push_back(nodeArgPtr->Name());
+  }
+  meta_def->inputs = inputs;
+  meta_def->outputs = outputs;
+  meta_def->since_version = 1;
+  meta_def->status = ONNX_NAMESPACE::EXPERIMENTAL;
+  sub_graph->SetMetaDef(std::move(meta_def));
+  sub_graph->nodes = sorted_nodes;
+  result.push_back(
+      std::make_unique<ComputeCapability>(std::move(sub_graph)));
+  return result;
+}
+
+common::Status TvmSoExecutionProvider::Compile(const std::vector<Node*>& nodes,
+                                               std::vector<NodeComputeInfo>& node_compute_funcs) {
+  printOptions();
+  for (auto* fused_node : nodes) {
+    auto func_body = fused_node->GetFunctionBody();
+    if (!func_body)
+      return common::Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Function body is empty");
+    // ORT_ENFORCE(checkHash(fused_node->ModelPath().ToPathString()),
```

Comment:
Will it be part of this PR?

Reply:
Thank for remark. It is a draft for handshake mechanism between onnx model and shared lib which should be prepared for it. I've added here TODO for more correct understanding. When API to get sha256 hash from shared lib is agreed and implemented it will be used here, but I think it will be slightly later. Basically I can transfer hash part to other PR, what do you think about it?

Reply:
That's fine with me. Another PR can change the location if needed.

Reply:
Up to you unless "slightly later" means something else.